source/vir/src/recursion.rs

```diff
@@ -431,8 +433,8 @@ pub(crate) fn check_termination_exp(
             let varx = ExpX::Var(unique_local(&str_ident(FUEL_PARAM)));
             let var_typ = Arc::new(TypX::Air(str_typ(FUEL_TYPE)));
             args.push(SpannedTyped::new(&exp.span, &var_typ, varx));
-            let rec_name = prefix_recursive_fun(&x);
-            let callx = ExpX::Call(CallFun::Fun(rec_name), typs.clone(), Arc::new(args));
+            let callx =
+                ExpX::Call(CallFun::CheckTermination(x.clone()), typs.clone(), Arc::new(args));
```

Comment:
@utaal This turned out not to be need by the commit, but it seems neater than the way this worked previously, so I left this change in.